src/sqlfluff/core/dialects/dialect_exasol_fs.py

```diff
@@ -0,0 +1,451 @@
+    BaseSegment,
+    Bracketed,
+    CommaDelimited,
+    Delimited,
+    GreedyUntil,
+    NamedSegment,
+    OneOf,
+    Ref,
+    ReSegment,
+    Sequence,
+    StartsWith,
+    SymbolSegment,
+)
+from .dialect_exasol import ObjectReferenceSegment, exasol_dialect
+
+exasol_fs_dialect = exasol_dialect.copy_as("exasol_fs")
+exasol_fs_dialect.sets("unreserved_keywords").add("ROWCOUNT")
+exasol_fs_dialect.set_lexer_struct(
+    [
+        (
+            "walrus_operator",
+            "regex",
+            r":=",
+            dict(is_code=True, type="walrus_operator"),
+        ),
+        (
+            "function_script_terminator",
+            "regex",
+            # TODO: this expression matches multiple functions within a raw string.
+            # but it only matches one script per raw string, because the statement terminator (;)
+            # is not required for scripts (e.g. not present in Python)
+            # need to find another solution but don't mess up with the divide operator.
+            r";\s+\/(?!\*)|\s+\/$",
+            dict(
+                is_code=True,
+                type="statement_terminator",
+                subdivide=dict(type="semicolon", name="semicolon", regex=r";"),
+                trim_post_subdivide=dict(
+                    type="newline", name="newline", regex=r"(\n|\r\n)*"
+                ),
+            ),
+        ),
+    ]
+    + exasol_fs_dialect.get_lexer_struct()
+)
+
+exasol_fs_dialect.add(
+    FunctionScriptTerminatorSegment=NamedSegment.make(
+        "function_script_terminator", type="statement_terminator"
+    ),
+    WalrusOperatorSegment=NamedSegment.make(
+        "walrus_operator", type="assignment_operator"
+    ),
+    VariableNameSegment=ReSegment.make(
+        r"[A-Z][A-Z0-9_]*",
+        name="function_variable",
+        type="variable",
+    ),
+)
+
+exasol_fs_dialect.replace(
+    SemicolonSegment=SymbolSegment.make(";", name="semicolon", type="semicolon"),
+)
+
+
+@exasol_fs_dialect.segment(replace=True)
+class StatementSegment(BaseSegment):
+    """A generic segment, to any of its child subsegments."""
+
+    type = "statement"
+
+    match_grammar = GreedyUntil(Ref("FunctionScriptTerminatorSegment"))
+    parse_grammar = OneOf(
+        Ref("CreateFunctionStatementSegment"),
+        Ref("CreateScriptingLuaScriptStatementSegment"),
+        Ref("CreateUDFScriptStatementSegment"),
+        Ref("CreateAdapterScriptStatementSegment"),
+    )
+
+
+@exasol_fs_dialect.segment(replace=True)
+class FileSegment(BaseSegment):
+    """This ovewrites the FileSegment from ANSI.
+
+    The reason is because SCRIPT and FUNCTION statements
+    are terminated by a trailing / at the end.
+    A semicolon is the terminator of the statement within the function / script
+    """
+
+    type = "file"
+    can_start_end_non_code = True
+    allow_empty = True
+    parse_grammar = Delimited(
+        Ref("StatementSegment"),
+        delimiter=Ref("FunctionScriptTerminatorSegment"),
```

Comment:
That's an interesting issue, I wonder if it's possible to change the parse grammar here to look for an EXASOL function segment first, and then fall back to a `;` delimited StatementSegment ?

Reply:
I tried it with `parse_grammar = OneOf(....)` but that doesn't work. It parsed the `StatementSegment` which was in the first place.